server/utils.go

```diff
@@ -173,3 +176,129 @@ func fixGithubNotificationSubjectURL(url string) string {
 func fullNameFromOwnerAndRepo(owner, repo string) string {
 	return fmt.Sprintf("%s/%s", owner, repo)
 }
+
+// filter lines in a string from start to end
+func filterLines(s string, start, end, limit int) string {
+	scanner := bufio.NewScanner(strings.NewReader(s))
+	var buf strings.Builder
+	i := 1
+	w := 1
+	for scanner.Scan() {
+		if w > limit {
+			break
+		}
+		if i < start {
+			i++
+			continue
+		}
+		if i > end {
+			break
+		}
+		buf.Write(scanner.Bytes())
+		buf.WriteByte(byte('\n'))
+		i++
+		w++
+	}
+	if err := scanner.Err(); err != nil {
+		fmt.Println(err)
+	}
+
+	return buf.String()
+}
+
+// getLineNumbers return the start and end lines from an anchor tag
+// of a github permalink.
+func getLineNumbers(s string) (start, end int) {
+	// split till -
+	parts := strings.Split(s, "-")
+
+	if len(parts) > 2 {
+		return -1, -1
+	}
+
+	switch len(parts) {
+	case 1: // just a single line
+		l := getLine(parts[0])
+		if l == -1 {
+			return -1, -1
+		}
+		if l < 3 {
+			return 0, l + 3
```

Comment:
@ali-farooq0 I think it makes sense to start from 0 so at least you get the first couple of lines before. For example, if l = 2, you get from the beginning of the file in the code lines and not start just from the line 2. This way is consistent as much possible with the case of `return l - 3, l + 3`

Reply:
@marianunez Hmm, fair enough, but then it's behaving differently for a rage where we return `start, end`. My thinking was that if someone wants to show a certain range they'll use that range i.e. `L13-L15`, but if they just have `L13` then the intent would be to show just that line.

I believe that's what github does as well.

<img width="624" alt="image" src="https://user-images.githubusercontent.com/25732808/66948537-e602ac80-f022-11e9-851b-eb709f974978.png">

link: https://github.com/mattermost/mattermost-server/pull/12390#discussion_r335118081

Reply:
We were following a different approach of showing the surrounding lines to give more context like it does in the code reviews screens. For example:

<img width="511" alt="Screen Shot 2019-10-16 at 8 08 17 PM" src="https://user-images.githubusercontent.com/936315/66967951-bff40100-f050-11e9-836d-c92f9a394076.png">

2/5 we should keep this because most of the time one line does not say enough.
However, we could also give the range the surrounding lines or limit the one liners to exactly one line to be consistent if you feel strongly about this.

Reply:
I agree with @marianunez about trying to show more context. Keeping it the same for now. Let me know if you want this changed.

Reply:
@marianunez @agnivade Alright lets keep it as is for now and we can always update this later if need be. 👌